### DISCUSSION TITLE
How do you use assert_result?

### DISCUSSION THREAD

**Discussion by batomow** [MikeSchulze/gdUnit3#190] [FAQ] [answered]
Im trying to test if a function runs without throwing any errors, but it tells me that i can't use 'is_success' on base nil. 
```
func test_add(
	cfuzzer := Fuzzers.rangei(-100 * cell_size, 100* cell_size), 
	rfuzzer := Fuzzers.rangei(0, cell_size-1)
	): 
	var base_cell = Vector2(cfuzzer.next_value(), cfuzzer.next_value())
	for __ in range(10): 
		var coords = base_cell + Vector2(rfuzzer.next_value(), rfuzzer.next_value())
		var n = Node2D.new()
		n.global_position = coords
		assert_result(hashmap.add(n)).is_success()
		auto_free(n)
```
![image](https://user-images.githubusercontent.com/15216737/147396374-8f3368e8-1bbf-4ffe-84d9-04a1d90ea7ea.png)
Is this not intended for function results ?

**Comment by MikeSchulze** [comment, marked as answer]
Hm, looks like the `hashmap.add(n)` returns null and the `assert_result.is_success()` can't handle null values.
You using the `Result` type as return type on `add(n)` right?

I will convert this question into a bug issue. https://github.com/MikeSchulze/gdUnit3/issues/191

Will be fixed by v1.1.2 where is release to day

Thanks for reporting.
